crates/matrix-sdk/src/room/mod.rs

```diff
@@ -3192,6 +3192,48 @@ impl Room {
     pub fn observe_live_location_shares(&self) -> ObservableLiveLocation {
         ObservableLiveLocation::new(&self.client, self.room_id())
     }
+
+    /// Mark a list of requests to join the room as seen, given their state
+    /// event ids.
+    pub async fn mark_join_requests_as_seen(&self, event_ids: &[OwnedEventId]) -> Result<()> {
```

Comment:
What if I still take the list of event ids and resolve the knock member events inside the method, failing if any of the provided event ids is not valid (doesn't exist or is not a knock member event) ?

Reply:
Yep, that could work too. It might be a bit suboptimal if the events haven't been cached yet, and we need to re-resolve them one way or another. Is there no way to pass the actual events here?

Reply:
There is, but it would mean having to find and deserialize the raw event to the right types in a couple of places instead of a single one. Another option could be to pass the `KnockJoinRequest` directly, maybe?

Reply:
Passing the `KnockRequest` could also work nicely, good idea!

Reply:
Ah good point that they're state events, so don't require the event cache \o/

Fine to pass the event ids, and check they match some known knock requests. If they don't, I'm not sure if we'd prefer to filter them out (i.e. ignore them), or return an error indicating so to the caller.

Reply:
I was going to use the event ids but then I realised `store.get_state_events_for_keys` uses the state_key which is the user id, not the event id. So maybe passing the user id is the best way to handle it: you got a knock request, have the user id as the state key, pass that to `mark_knock_requests_as_seen`, if the state event is in knock state, we mark it as seen.